code/game/objects/items/stacks/stack.dm

```diff
@@ -55,11 +55,11 @@
 	. = ..()
 	if(!dynamic_icon_state)
 		return
-	var/temp_amount = get_amount()
-	if(temp_amount > 1)
-		icon_state = "[initial(icon_state)]_[min(temp_amount, 3)]" //2 if amount is 2, 3 if more.
+	var/state = CEILING((amount/max_amount) * 3, 1)
+	icon_state = "[initial(icon_state)]_[state]"
```

Comment:
You should probably put this below the `return` so that you're not unnecessarily changing the icon_state twice. If you keep it this way, please remove the hanging return on line 62, as its not doing anything otherwise. Looks good to me otherwise